src/app/booking/OOSPlannerSection.js

```diff
@@ -415,8 +419,59 @@ function FriendsBookings({ childrenData = [], weekDates }) {
   );
 }
 
+export const ShareCalendar = ({ childrenData, showSharePopup }) => {
+
+  return (
+    <>
+      {childrenData?.length === 1 ? (
+        <Button startIcon={<IosShareIcon />} variant="outlined" color="grey" onClick={() => {
+          showSharePopup(childrenData[0].id);
+        }}>
+          Share Calendar
+        </Button>
+      ) : (
+        <PopupState variant="popover" popupId="children-popup-menu">
+          {(popupState) => (
+            <>
+              <Button startIcon={<IosShareIcon />} {...bindTrigger(popupState)} variant="outlined" color="grey">
+                  Share Calendar
+              </Button>
+              <Menu
```

Comment:
Why do not extract it to a separate component, since it is repeated in the system?